[CI] StatelessReshardDisruptionIT testReshardWithDisruption failing

**Build Scans:**
- [elasticsearch-intake #44600 / part2](https://gradle-enterprise.elastic.co/s/2osnshn35abjm)
- [elasticsearch-pull-request #166797 / part-2](https://gradle-enterprise.elastic.co/s/o3ua3uajovq36)
- [elasticsearch-pull-request #166722 / part-2](https://gradle-enterprise.elastic.co/s/vzug7aqelikye)
- [elasticsearch-pull-request #166699 / part-2](https://gradle-enterprise.elastic.co/s/ae2wldmzrfybc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.reshard.StatelessReshardDisruptionIT.testReshardWithDisruption" -Dtests.seed=DA8573BF059EE114 -Dtests.locale=ku-Latn-IQ -Dtests.timezone=America/Havana -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.reshard.StatelessReshardDisruptionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testReshardWithDisruption'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: timed out waiting for green state
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testReshardWithDisruption
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 4 failures in test testReshardWithDisruption (1.0% fail rate in 393 executions)
- [main] 3 failures in step part-2 (1.4% fail rate in 211 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.4% fail rate in 208 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

:eyes:

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -589,9 +589,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecStatsFirstLastIT
   method: test {csv-spec:stats_first_last.Test aggregating on integers from row}
   issue: https://github.com/elastic/elasticsearch/issues/153700
-- class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardDisruptionIT
-  method: testReshardWithDisruption
-  issue: https://github.com/elastic/elasticsearch/issues/155150
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecStatsIT
   method: test {csv-spec:stats.sumWithOverflowGroupingRow}
   issue: https://github.com/elastic/elasticsearch/issues/153434

